cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/StatementHandler.kt

```diff
@@ -83,6 +84,26 @@ class StatementHandler(frontend: PythonLanguageFrontend) :
         }
     }
 
+    /**
+     * Translates a Python [`Delete`](https://docs.python.org/3/library/ast.html#ast.Delete) into a
+     * [DeleteExpression].
+     */
+    private fun handleDelete(node: Python.AST.Delete): DeleteExpression {
+        val delete = newDeleteExpression(rawNode = node)
+        node.targets.forEach { target ->
+            if (target is Python.AST.Subscript) {
+                delete.targets.add(frontend.expressionHandler.handle(target))
+            } else {
+                delete.additionalProblems +=
+                    newProblemExpression(
+                        problem = "Delete is only supported for Subscript targets.",
```

Comment:
Why? According to the Python doc: "`targets` is a list of nodes, such as [Name](https://docs.python.org/3/library/ast.html#ast.Name), [Attribute](https://docs.python.org/3/library/ast.html#ast.Attribute) or [Subscript](https://docs.python.org/3/library/ast.html#ast.Subscript) nodes." -> I think we should support Name and Attribute as well. Or is there a good reason not to?

Reply:
The issue is that we cannot fully support the semantics of deleting a name or an attribute, since it un-declares an established symbol (we don't even have an API for that in the scope manager). For subscript nodes, it deletes content inside a list or dict and we can potentially model this with DFG nodes (and in the future maybe do some value approximation). I would suggest to clarify this in the problem expression.

Reply:
Added additional information to the problem expression message.